src/test/test_btf.cpp

```diff
@@ -71,3 +72,78 @@ BTF_CASE(tail_call_bad)
 BTF_CASE(twomaps)
 BTF_CASE(twostackvars)
 BTF_CASE(twotypes)
+
+/**
+ * @brief Write a type into a vector and grow it.
+ *
+ * @tparam T Type to write
+ * @param[in] btf The .BTF section
+ * @param[in] value The value to write
+ */
+template <typename T>
+void _write_btf(std::vector<uint8_t>& btf, T& value) {
+    size_t length = 0;
+    size_t offset = btf.size();
+    if constexpr (std::is_same<T, std::string>::value) {
+        length = value.length();
+        btf.resize(offset + length + 1);
+        memcpy(btf.data() + offset, value.c_str(), length + 1);
+    } else {
+        length = sizeof(T);
+        btf.resize(offset + length);
+        memcpy(btf.data() + offset, &value, length);
+    }
+}
+
+void _write_btf_header(std::vector<uint8_t>& btf) {
+    size_t offset = 0;
+    btf_header_t header = {};
+    header.magic = BTF_HEADER_MAGIC;
+    header.version = BTF_HEADER_VERSION;
+    header.hdr_len = sizeof(btf_header_t);
+    header.type_off = 0;
+    header.type_len = 0;
+    _write_btf(btf, header);
+}
+
+TEST_CASE("validate-parsing-simple-loop", "[BTF]")
+{
+    std::vector<uint8_t> btf;
+    btf_type_t type = {};
+
+    _write_btf_header(btf);
+
+    type.info = BTF_KIND_PTR << 24;
+    type.type = 1;
+    _write_btf(btf, type);
```

Comment:
I think it can be `_write_btf(btf, btf_type_t{.info = BTF_KIND_PTR << 24, .type = 1);`, or decalre and initialize type just outside the call. It's more importand in the next test, where you can avoid reusing a variable.